lib/find/find.js

```diff
@@ -106,6 +107,9 @@ function urlToPath(value, config) {
   }
 
   try {
+    // node 8 support
+    URL = require('url').URL
```

Comment:
Would we want to move this into the catch, using global `URL` by default?

Reply:
Yeah, would be good if `var URL = require('url').URL` was moved to the top requires

Reply:
I think these are conflicting statements. Should I move it to global or to the catch?

Reply:
Sorry yeah, those are contradicting.
`remark-validate-links` is mostly used in Node, so it doesn’t matter a lot to require `url`.
I think moving `require` to the top is a good idea.

@ChristianMurphy what’s the reason for first trying the global URL, and then falling back to `require('url').URL`?

Reply:
Node js has two `Url` implementations.
the global `Url` which follows the WHATWG standard. https://nodejs.org/api/url.html#url_the_whatwg_url_api
and `require('url').Url` which deviates from the standard and is deprecated, but is supported in Node 8 and before. https://nodejs.org/api/url.html#url_legacy_url_api

Given that `require('url').Url` is deprecated and non-standard, I'd prefer not to use it at all, but would would accept using it as a fallback for Node 8 compatability.

Reply:
In my understanding, yes, the *rest* of the `require('url')` API is deprecated, but `require('url').URL` is the same as the global `URL`, and it has better support:

```js
> require('url').URL === URL
true
```

This is also in line with the current docs:

> Browser-compatible URL class, implemented by following the WHATWG URL Standard. Examples of parsed URLs may be found in the Standard itself. The URL class is **also** available on the global object.
> https://nodejs.org/api/url.html#url_class_url

(emphasis mine). The rest of the docs no longer reference `require('url').URL`, but that `also` (and my experience) indicates it works just as well.

Reply:
Ah, there’s also a `.Url`, I didn’t know that, and it wasn’t what I referred to. Confusing.

Reply:
> Ah, there’s also a `.Url`, I didn’t know that, and it wasn’t what I referred to. Confusing.

Thanks for the clarification.
I'd still lean toward preferring global `URL` over `require('url').URL` because WebPack/Rollup may try to polyfill `require('url').URL` due to it being a `require` but should leave `URL` alone since it is a global browser. (:notebook: this is based off past experience with Globals vs Modules, bundlers may have more intelligent mappings in recent releases)

Reply:
I moved it to the top. Is that good?